src/tag_util.h

```diff
@@ -8,9 +8,30 @@
 
 #include "tags.h"
 
+/**
+ * At the time of writing older clang 6 lacks support for template
+ * argument deduction of class templates 
+ * http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0620r0.html
+ * https://clang.llvm.org/cxx_status.html
+ *
+ * This make helper uses function template argument deduction to solve
+ * for std::array declarations.
+ */
+template <typename... T>
+constexpr auto make_array(T&&... values) ->
+    std::array<
+       typename std::decay<
+           typename std::common_type<T...>::type>::type,
```

Comment:
Use C++14's `std::decay_t` and `std::common_type_t` to avoid the `typename ... :: type` verbosity.

Reply:
👍